juju/apiclient.py

```diff
@@ -17,10 +25,250 @@
     websocket.create_connection
 except AttributeError:
     raise RuntimeError(
-        "Expected 'python-websocket' "
+        "Expected 'python-websocket' package or 'websocket-client' from pypi, "
         "found incompatible gevent 'websocket'")
 
 
 websocket.logger = logging.getLogger("websocket")
 logger = logging.getLogger("juju")
 
+
+class UnknownFacade(Exception):
+    """UnknownFacade is raised if the server doesn't have the requested facade.
+
+    This can occur if asking for a non-existent facade, or a newer facade that
+    is not supported on the API server, or even a known facade that isn't
+    supported at the endpoint that has been connected to.
+
+    """
+    def __init__(self, *args):
+        super(UnknownFacade, self).__init__(*args)
+
+
+class FacadeVersionNotSupported(Exception):
+    """FacadeVersionNotSupported is raised if the version requested isn't supported.
+
+    """
+    def __init__(self, *args):
+        super(UnknownFacade, self).__init__(*args)
+
+
+class ServerError(Exception):
+    """Base exception class for particular server side errors.
+
+    If there is no specific error type for the error code returned from the
+    server, then this is the type of the returned error.
+
+    """
+    def __init__(self, error_code, message):
```

Comment:
For Python 2, we might want to call the base class initializer, e.g. passing the message.
See http://bugs.python.org/issue1692335